### PR TITLE
Fixes Issue 2308

### LINKED ISSUE #2308: [DAM analyzer] Spurious warnings for typeof(Foo) arguments
```csharp
public static void Main() {
    M(typeof(C));
}

// warning IL2082: 'type' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicMethods' in call to 'C.M(Type)'.
// The implicit 'this' argument of method 'C.Main()' does not have matching annotations.
// The source value must declare at least the same requirements as those declared on the target location it is assigned to.
private static void M([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)] Type type) {}
```

@mateoatr I think the typeof case here is causing the extra warning:
https://github.com/dotnet/linker/blob/83c51d44e0ec5cdf408f93bc5576cc22e9cc6ced/src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs#L216

I think the intention was to support typeof(T) flowing into an annotated location where T references a generic parameter. Unfortunately it produces unexpected warnings in cases like the above. Discovered while adding shared tests as part of https://github.com/dotnet/linker/pull/2307.

### PR DESCRIPTION
**Pull request by LakshanF**
Fixes #2308 by trying to get the symbol for an operation type only if it is generics

### CODE REVIEW

**Comment by tlakollo** [comment]
@sbomer I see that the lint and some of the other ci tasks are failing because they are trying to run in Ubuntu 16
`##[warning]An image label with the label Ubuntu16 does not exist.`
I think this was solving it https://github.com/dotnet/linker/pull/2338
But it hasn't been merged in the DAM branch, right?

File: test/Mono.Linker.Tests.Cases/DataFlow/MethodReturnParameterDataFlow.cs [hunk 1]
```diff
@@ -50,11 +50,6 @@ static Type NoRequirements ()
 		[UnrecognizedReflectionAccessPattern (typeof (MethodReturnParameterDataFlow), nameof (ReturnPublicParameterlessConstructor),
 			new Type[] { typeof (Type), typeof (Type), typeof (Type) }, returnType: typeof (Type), messageCode: "IL2068")]
 		// TODO: https://github.com/dotnet/linker/issues/2308
```
Comment (sbomer) [inline test/Mono.Linker.Tests.Cases/DataFlow/MethodReturnParameterDataFlow.cs:52]: Nit: remove the TODOs here and below please

**Review by sbomer — APPROVED** [review]
Looks good, thank you!

**Pull request merged**